src/graphqlProtocol/action.ts

```diff
@@ -99,7 +99,7 @@ export const handlePermission = async (
     where: permissionWhere,
     lookupPermissionEntity,
   } = await buildActionPermissionFilter(
-    permission,
+    () => permission,
```

Comment:
`permission` could be a permission object or a function returning a permission. 
The additional function wrapper will break the tests if a permission function is provided.

To be honest, `buildActionPermissionFilter` needs a serious review. 
For example, converting the params into a single object. 
Maybe then the type issue will be easier to solve.

Reply:
buildActionPermissionFilter expects a function as first parameter tho, I'll implement a type check and wrap it into a function if it is an object so that it doesn't break. would be the best solution for now i think

Reply:
i've cleaned up the types a bit so that it is in line with what's actually stored and returned